### PR TITLE
Case 19660: Fix HUD crash

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/19660/std-function-void-gpu-Batch-std-shared_ptr-gpu-Texture-11a119c88dc0c455ca3ff849cca62615944e29027bd9f4652fe8f80beb42aa66

Test plan:
- No idea what causes this crash.
- In desktop mode, the HUD should render (toolbar, audio meter, etc.)
- In HMD, the HUD sphere should render (audio meter, stats).
- Switch between modes and you shouldn't crash.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15292/HighFidelity-Beta-PR14443-5840849.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15292/HighFidelity-Beta-PR14443-5840849.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15292/HighFidelity-Beta-Interface-PR14443-5840849.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15293/HighFidelity-Beta-PR14443-5840849.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15293/HighFidelity-Beta-PR14443-5840849.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15293/HighFidelity-Beta-Interface-PR14443-5840849.dmg

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15294/HighFidelity-Beta-PR14443-5840849.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15294/HighFidelity-Beta-PR14443-5840849.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15294/HighFidelity-Beta-Interface-PR14443-5840849.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15295/HighFidelity-Beta-PR14443-5840849.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15295/HighFidelity-Beta-PR14443-5840849.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15295/HighFidelity-Beta-Interface-PR14443-5840849.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15308/HighFidelity-Beta-PR14443-a618ea6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15308/HighFidelity-Beta-Interface-PR14443-a618ea6.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15311/HighFidelity-Beta-PR14443-101b45d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15311/HighFidelity-Beta-PR14443-101b45d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15311/HighFidelity-Beta-Interface-PR14443-101b45d.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15312/HighFidelity-Beta-PR14443-30ce100.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15312/HighFidelity-Beta-PR14443-30ce100.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15312/HighFidelity-Beta-Interface-PR14443-30ce100.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15323/HighFidelity-Beta-PR14443-c99a810.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15323/HighFidelity-Beta-PR14443-c99a810.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15323/HighFidelity-Beta-Interface-PR14443-c99a810.exe

File: libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp [hunk 1]
```diff
@@ -420,18 +420,26 @@ void HmdDisplayPlugin::HUDRenderer::updatePipeline() {
...
-            gpu::BufferView posView(vertices, VERTEX_OFFSET, vertices->getSize(), VERTEX_STRIDE, format->getAttributes().at(gpu::Stream::POSITION)._element);
-            gpu::BufferView uvView(vertices, TEXTURE_OFFSET, vertices->getSize(), VERTEX_STRIDE, format->getAttributes().at(gpu::Stream::TEXCOORD)._element);
+    auto hudPipeline = pipeline;
+    auto hudFormat = format;
+    auto hudVertices = vertices;
+    auto hudIndices = indices;
+    auto hudUniformBuffer = uniformsBuffer;
+    auto hudUniforms = uniforms;
+    auto hudIndexCount = indexCount;
+    return [=](gpu::Batch& batch, const gpu::TexturePointer& hudTexture, bool mirror) {
```
Comment (sabrina-shanman) [inline libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: Not a blocker, but I think explicit capture is a bit more robust.

Comment (SamGondelman) [reply libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: Normally I would totally agree, but this is very self-contained and writing them all out was a mess.

Comment (sabrina-shanman) [reply libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.cpp]: Fair enough.

**Comment by qaAdamSmith82** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15626/HighFidelity-Beta-PR14443-c99a810.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15626/HighFidelity-Beta-PR14443-c99a810.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15626/HighFidelity-Beta-Interface-PR14443-c99a810.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15626/HighFidelity-Beta-PR14443-c99a810.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15626/HighFidelity-Beta-Interface-PR14443-c99a810.exe

**Comment by qaAdamSmith82** [comment]
Pre-Merge Tests completed

**Pull request merged**